### PR TITLE
Fix `BlockId` collecting

### LINKED ISSUE #1057: Incorrect `BlockId` syntax in attributes passes compilation
E.g. `BlockId::Number(Latest)` -> runner treats it as `BlockId::Tag(Latest)`

Other attributes should checked for this kind of behaviour too and, if needed, appropriate issues should be created

### PR DESCRIPTION
**Pull request by cptartur**
Closes #1057

## Introduced changes

-

## Checklist

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by piotmag769 — COMMENTED** [review]
Can we put the tests in a single file though?

**Comment by cptartur** [comment]
> Can we put the tests in a single file though?

I don't think so. It seems like if there are multiple tests with invalid attribute in a file, it will only output one.

**Comment by cptartur** [comment]
> > Can we put the tests in a single file though?
> 
> I don't think so. It seems like if there are multiple tests with invalid attribute in a file, it will only output one.

Never mind, it seems to work

File: crates/test-collector/src/lib.rs [hunk 1]
```diff
@@ -333,49 +334,51 @@ fn extract_fork_config_from_args(db: &dyn SyntaxGroup, attr: &Attribute) -> Opti
         .identifier(db)
```
[the remark this answers is not in the document]
Comment (cptartur) [reply crates/test-collector/src/lib.rs:334]: What exactly should be asserted?

[the remark this answers is not in the document]
Comment (piotmag769) [reply crates/test-collector/src/lib.rs:334]: That the identifier starts with `"BlockId"`, currently sth like `block_id: Number(123)` will work

File: crates/forge/tests/data/diagnostics_and_plugins/tests/test_fork.cairo [hunk 2]
```diff
@@ -52,3 +52,32 @@ fn incorrect_fork_attributes9() {
...
+#[fork(url: "http://188.34.188.184:9545/rpc/v0.4", block_id: BlockWhat::Number(12345))]
+fn incorrect_fork_attributes13() {
+    assert(1 == 1, 'ok')
+}
+
+#[test]
+#[fork(url: "http://188.34.188.184:9545/rpc/v0.4", block_id: Something::BlockId::Number(12345))]
+fn incorrect_fork_attributes14() {
+    assert(1 == 1, 'ok')
+}
```
Comment (piotmag769) [inline crates/forge/tests/data/diagnostics_and_plugins/tests/test_fork.cairo:83]: newline

**Review by piotmag769 — APPROVED** [review]

File: docs/src/testing/fork-testing.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ It is possible to pass `url` and `block_id` arguments to the `fork` attribute:
 enum BlockId {
     Tag: BlockTag,
     Hash: felt252,
-    Number: felt252,
+    Number: u64,
```
Comment (piotmag769) [inline docs/src/testing/fork-testing.md:24]: it should be changed in snforge_std too

**Review by drknzz — APPROVED** [review]

**Pull request merged**